specs/Standards/MultiToken/README.md

```diff
@@ -0,0 +1,539 @@
+    /// * `token_ids`: the `token_ids` argument given to `mt_transfer_call`
+    /// * `approvals`: if using Approval Management, contract MUST provide
+    ///   set of original approved accounts in this argument, and restore these
+    ///   approved accounts in case of revert. In this case it may be multiple sets of approvals
+    ///
+    /// Returns true if tokens were successfully transferred to `receiver_id`.
+    fn mt_resolve_transfer(
+        &mut self,
+        sender_id: AccountId,
+        receiver_id: AccountId,
+        token_ids: Vec<TokenId>,
+        amounts: Vec<U128>,
+    ) -> Vec<U128>;
+}
+```
+### Storage Management Trait
+#### Notes
+This is semi necessary for ft token types to be able to refund users for storage of many different token types like gold/silver... this might be slightly out of scope
+```
+pub trait StorageManagement {
+    // if `registration_only=true` MUST refund above the minimum balance if the account didn't exist and
+    //     refund full deposit if the account exists.
+    fn storage_deposit(
+        &mut self,
+        token_ids: Vec<TokenId>,
+        account_id: Option<AccountId>,
+        registration_only: Option<bool>,
+    ) -> StorageBalance;
+
+    /// Withdraw specified amount of available Ⓝ for predecessor account.
+    ///
+    /// This method is safe to call. It MUST NOT remove data.
+    ///
+    /// `amount` is sent as a string representing an unsigned 128-bit integer. If
+    /// omitted, contract MUST refund full `available` balance. If `amount` exceeds
+    /// predecessor account's available balance, contract MUST panic.
+    ///
+    /// If predecessor account not registered, contract MUST panic.
+    ///
+    /// MUST require exactly 1 yoctoNEAR attached balance to prevent restricted
+    /// function-call access-key call (UX wallet security)
+    ///
+    /// Returns the StorageBalance structure showing updated balances.
+    fn storage_withdraw(&mut self, token_ids:Vec<TokenId>, amount: Option<U128>) -> StorageBalance;
+
+    /// Unregisters the predecessor account and returns the storage NEAR deposit back.
+    ///
+    /// If the predecessor account is not registered, the function MUST return `false` without panic.
+    ///
+    /// If `force=true` the function SHOULD ignore account balances (burn them) and close the account.
+    /// Otherwise, MUST panic if caller has a positive registered balance (eg token holdings) or
+    ///     the contract doesn't support force unregistration.
+    /// MUST require exactly 1 yoctoNEAR attached balance to prevent restricted function-call access-key call
+    /// (UX wallet security)
+    /// Returns `true` iff the account was unregistered.
+    /// Returns `false` iff account was not registered before.
+    fn storage_unregister(&mut self, token_ids:Vec<TokenId>, force: Option<bool>) -> Vec<bool>;
+
+    fn storage_balance_bounds(&self, token_id:Vec<TokenId>, account_id: Option<AccountId>) -> StorageBalanceBounds;
```

Comment:
What is this function for? Seems like it is missing a description?

Reply:
// Returns minimum and maximum allowed balance amounts to interact with this
// contract given  specified token_ids.